include/ck/tensor_operation/gpu/device/impl/device_elementwise_normalization_impl.hpp

```diff
@@ -0,0 +1,591 @@
+    const GridDesc_M_K y_grid_desc_m_k,                     // Descriptor of Y
+    index_t num_k_block_tile_iteration,                     // arg.numBlockTileIteration_
+    AccDataType epsilon,                                    // Datatype of epsilon
+    const InDataTypePointerTuple p_in_global_tuple,         // Ptr tuple of input matrixs
+    CDataType* const __restrict__ p_c_global,               // Ptr of C
+    const GammaDataType* const __restrict__ p_gamma_global, // Ptr of gamma
+    const BetaDataType* const __restrict__ p_beta_global,   // Ptr of beta
+    YDataType* const __restrict__ p_y_global,               // Ptr of y
+    const ElementwiseOperation elementwise_op,              // Operation Add
+    const AccElementwiseOperation acc_elementwise_op)       // Operation Passthrough
+{
+    GridwiseElementwiseReduction::Run(in_grid_2d_desc_tuple,      // Descriptor tuple of A & B
+                                      c_grid_desc_m_k,            // Descriptor of C
+                                      gamma_grid_desc_m_k,        // Descriptor of Gamma
+                                      beta_grid_desc_m_k,         // Descriptor of Beta
+                                      y_grid_desc_m_k,            // Descriptor of Y
+                                      num_k_block_tile_iteration, // arg.numBlockTileIteration_
+                                      epsilon,                    // epsilon
+                                      p_in_global_tuple,          // Ptr tuple of A & B
+                                      p_c_global,                 // Ptr of C
+                                      p_gamma_global,             // Ptr of gamma
+                                      p_beta_global,              // Ptr of beta
+                                      p_y_global,                 // Ptr of Y
+                                      elementwise_op,             // Add
+                                      acc_elementwise_op);        // Passthrough
+};
+} // namespace ck
+
+namespace ck {
+namespace tensor_operation {
+namespace device {
+
+// Y = LayerNorm(A + B, Beta, Gamma)
+template <typename InDataTypeTuple, // Datatype of A & B
+          typename GammaDataType,   //
+          typename BetaDataType,
+          typename AccDataType,
+          typename YDataType,
+          typename ElementwiseOperation,
+          typename AccElementwiseOperation,
+          index_t Rank,               //
+          index_t NumReduceDim,       //
+          index_t BlockSize,          //
+          index_t MThreadClusterSize, // Num of threads in a block on M direction
+          index_t KThreadClusterSize, // Num of threads in a block on N direction
+          index_t MThreadSliceSize,   // Each thread calculate rows
+          index_t KThreadSliceSize,   // Each thread calculate columns
+          index_t XYSrcVectorDim,     // Dimension to do reduce
+          index_t XSrcVectorSize,     // Size to fetch source x
```

Comment:
What is X here?
Do you mean InDataTypeTuple?
We need to define a consistent naming for input tensor tuple. (In is also OK)

Reply:
Yes, you are right, the naming is confused.
Really thank you for your advice.
I'll make it more reasonable.